pos_margin/models/pos_order_line.py

```diff
@@ -33,17 +29,34 @@ def _compute_multi_margin(self):
         for line in self.filtered("product_id"):
             purchase_price = self._get_purchase_price(line)
             line.purchase_price = purchase_price
-            line.margin = line.price_subtotal - (purchase_price * line.qty)
+            if purchase_price:
+                line.margin = line.price_subtotal - (purchase_price * line.qty)
 
     @api.model
     def _get_purchase_price(self, line):
-        # We call _get_purchase_price from sale_margin module, to reuse
-        # computation that handles multi currencies context and UoM
-        SaleOrderLine = self.env["sale.order.line"]
-
-        # if used in combination with module which does add the uom_id to line
-        uom = hasattr(line, "uom_id") and line.uom_id or line.product_id.uom_id
-
-        return SaleOrderLine._get_purchase_price(
-            line.order_id.pricelist_id, line.product_id, uom, line.order_id.date_order
-        )["purchase_price"]
+        if not line.product_id:
```

Comment:
Hi, as the following lines are a copy past from odoo code, could you : 
- add a comment with a link to the copied code (https://github.com/odoo/odoo/blob/14.0/addons/sale_margin/models/sale_order.py#L20)
- add credits to Odoo for this file ?
-

Reply:
Correct me if I'm wrong, but can't we drop the dependency to `sale_margin` now that the method has been copied here? IF so also update `ROADMAP.rst`